compiler-core/src/build.rs

```diff
@@ -99,8 +99,11 @@ pub struct Package {
 
 impl Package {
     pub fn attach_doc_and_module_comments(&mut self) {
+        let hidden_modules = self.specify_hidden_modules();
         for mut module in &mut self.modules {
-            module.attach_doc_and_module_comments();
+            if !hidden_modules.contains(&module.name) {
+                module.attach_doc_and_module_comments();
+            }
```

Comment:
We still want to attach documentation for hidden modules as it's not only used by the HTML documentation generator, so best revert this